comicapi/genericmetadata.py

```diff
@@ -202,24 +252,129 @@ def get_clean_metadata(self, *attributes: str) -> GenericMetadata:
+            elif isinstance(new, (list, set)) and len(new) == 0:
+                pass
+            else:
+                setattr(self, cur, new)
+
+    def assign_overwrite(self, cur: str, new: Any) -> None:
+        """Overwrite - Overwrite everything!"""
+        # TODO type check?
+        setattr(self, cur, new)
+
+    def assign_overwrite_no_clear(self, cur: str, new: Any) -> None:
+        """Overwrite (no clear/reset) - Same as Overwrite but None/empty string from 'new' do NOT overwrite values in 'self'."""
+        if new is not None:
+            if isinstance(new, (list, set)) and len(new) > 0 and isinstance(getattr(self, cur), (list, set)):
+                setattr(self, cur, new)
+            elif isinstance(new, str) and len(new) > 0:
+                setattr(self, cur, new)
+            elif isinstance(new, (int, float, bool)):
+                setattr(self, cur, new)
+
+    def assign_add_missing(self, cur: str, new: Any) -> None:
+        """Add Missing - Any values that are None or an empty str/list/set, add them from 'new' to 'self'."""
+        if new is not None:
+            if (
+                isinstance(getattr(self, cur), (list, set))
+                and len(getattr(self, cur)) == 0
+                and isinstance(new, (list, set))
+                and len(new) > 0
+            ):
+                setattr(self, cur, new)
+            elif (
+                isinstance(getattr(self, cur), str)
+                and len(getattr(self, cur)) == 0
+                and isinstance(new, str)
+                and len(new) > 0
+            ):
+                setattr(self, cur, new)
+            elif getattr(self, cur) is None:
+                setattr(self, cur, new)
+
+    def assign_combine_new(self, cur: str, new: Any) -> None:
+        """Combine (New) - Combine lists and sets. Replace all None values. Any other values that collide the 'new' value is used (ignoring None and empty strings)."""
+        if new is not None:
+            if isinstance(new, (list, set)) and isinstance(getattr(self, cur), (list, set)):
+                self.assign_dedupe(new, cur, True)
+            elif (
+                isinstance(new, str)
+                and len(new) > 0
+                and (isinstance(getattr(self, cur), str) or getattr(self, cur) is None)
+            ):
+                setattr(self, cur, new)
+            else:
+                # int, float and bool
+                setattr(self, cur, new)
+
+    def assign_combine_cur(self, cur: str, new: Any) -> None:
+        """Combine (Current) - Same as Combine (New) but 'self' value is kept with collisions."""
+        if new is not None:
+            if isinstance(new, (list, set)) and isinstance(getattr(self, cur), (list, set)):
+                self.assign_dedupe(new, cur)
+            elif isinstance(new, str) and len(getattr(self, cur)) == 0:
+                setattr(self, cur, new)
+            elif getattr(self, cur) is None:
+                setattr(self, cur, new)
```

Comment:
Why are there 2 of these. The only list/set type you account for is a str (because we don't have any others currently) but that means they are all strings! The only thing you are preserving is case. As named this should have the same implementation as overlay but instead of replacing a list it merges them `if new not in getattr(self, 'cur'): getattr(self, 'cur').add(new)` and `if new not in getattr(self, 'cur'): getattr(self, 'cur').append(new)`

Reply:
I've cleaned this up now and use a `dict` to dedupe so it should be clearer.